### ISSUE TITLE
Show labels using 4 color theorem

### ISSUE THREAD

**Issue by koopa31** [napari/napari#6450]
## 🚀 Feature
Give the possibility to show labels colors using 4 color theorem (https://en.wikipedia.org/wiki/Four_color_theorem)

## Motivation

This is really convenient when two touching objects with different labels are randomly given the same color which makes think a segmentation failed.

## Pitch

Would it be possible to give the possibility to use this representation? An implementation of this theorem is already available at: https://pypi.org/project/ncolor/. So I think it would not be very costly to add as a feature.

**Comment by psobolewskiPhD** [comment]
> This is really convenient when two touching objects with different labels are randomly given the same color which makes think a segmentation failed.

I don't think this should happen anymore after 0.4.19 is released.
Still labels could have *similar* colors which would make discerning objects difficult.
So I like this idea quite a bit as a toggle-able mode?

**Comment by koopa31** [comment]
Yes the idea would be to switch between the two modes, so users can choose.

**Comment by Czaki** [comment]
> I don't think this should happen anymore after 0.4.19 is released.

I think that it still could happen. 0.4.19 will resolve consistency between the color picker and the color visible on the canvas. It will not solve the problem of casing the same color for two touching objects. 


> An implementation of this theorem is already available at: [pypi.org/project/ncolor](https://pypi.org/project/ncolor/). So I think it would not be very costly to add as a feature.

for many use cases, this steep will be really costly. And will be not possible to apply to chunked data. 

This issue is partially related to #6449

**Comment by kevinjohncutler** [comment]
@Czaki ncolor is very fast on typical label matrices, but I agree that it should be compatible with chunked data if it were to be integrated into napari. I'm open to pull requests on that front if anyone wants to work on it. That said, @koopa31, it might make more sense to make a plugin for ncolor instead.

**Comment by Czaki** [comment]
@kevinjohncutler Because of performance reasons, we do not even calculate min/max on enough big data. 
The algorithm that will reconstruct the connection graph cannot be faster. 

Also, it may happen that there will be few distinct. component with a single number that breaks the assumptions of 4 color theorem.

**Comment by jni** [comment]
It's funny, I was just thinking about this yesterday. But, another issue is that in napari we deal with multidimensional images, which breaks the four color assumption — for example, if you are tracking cells, they might not touch in one frame but then touch in a later frame.

My instinct is the same as @kevinjohncutler — this functionality could/should go in a plugin, which would take in a labels layer, apply the 4-color theorem, and then apply direct color mode to the layer with the computed colors. (And I'll note that we do plan for plugins to be able to contribute to menus, potentially including right-click menus, in the near future, so this plugin would be much more easily accessible.)

After fixing #6448 (will happen before 0.4.19), it should be possible to check on touching labels after 1-2 shuffles.

Given all this, and the fact that we have a lot of work to do optimising labels rendering in general, I'm inclined to close this as wontfix... But I am happy to leave it open for a bit for further discussion.
